### ISSUE TITLE
Build fails because of unrecognized argument passed to virtualenv

### ISSUE THREAD

**Issue by jack1142** [readthedocs/readthedocs.org#6725]
## Details

* Read the Docs project URL: https://readthedocs.org/projects/red-discordbot/
* Build URL (if applicable): https://readthedocs.org/projects/red-discordbot/builds/10523539/
* Read the Docs username (if applicable): jack1142

## Expected Result

Build should be successful.

## Actual Result

Build failed on creating virtualenv step:

```
[rtd-command-info] start-time: 2020-02-28T18:50:40.863059Z, end-time: 2020-02-28T18:50:41.986144Z, duration: 1, exit-code: 2
python3.8 -mvirtualenv --no-site-packages /home/docs/checkouts/readthedocs.org/user_builds/red-discordbot/envs/latest
usage: virtualenv [--version] [--with-traceback] [-v | -q] [--app-data APP_DATA] [--clear-app-data] [--discovery {builtin}] [-p py] [--creator {builtin,cpython3-posix,venv}] [--seeder {app-data,pip}] [--no-seed]
                  [--activators comma_sep_list] [--clear] [--system-site-packages] [--symlinks | --copies] [--download | --no-download] [--extra-search-dir d [d ...]] [--pip version] [--setuptools version] [--wheel version] [--no-pip]
                  [--no-setuptools] [--no-wheel] [--symlink-app-data] [--prompt prompt] [-h]
                  dest
virtualenv: error: unrecognized arguments: --no-site-packages
```

**Comment by gvanrossum** [comment]
I see this too for mypy (https://github.com/python/mypy/issues/8459)

**Comment by jack1142** [comment]
I forgot to post an update but on our project we only really got it at the build I linked and [one other](https://readthedocs.org/projects/red-discordbot/builds/10523512/) that was ran around same time, it seems like it fixed itself later.

**Comment by gvanrossum** [comment]
I still think this requires some fix in readthedocs. [This code](https://github.com/readthedocs/readthedocs.org/blob/master/readthedocs/doc_builder/python_environments.py#L293) passes either `--no-site-packages` or `--system-site-packages`. But (as of now -- I don't know since when) virtualenv doesn't have a `--no-site-packages` flag, so readthedocs shouldn't be passing it.

As to why it sometimes works and sometimes doesn't, maybe it's a recent change to virtualenv and this code path is only taken for fresh VMs/containers, or something.

**Comment by gvanrossum** [comment]
Update: this disappeared (without specific mention in the ChangeLog) in virtualenv 20.0.0.

**Comment by gaborbernat** [comment]
Hello, you can still find the legacy (v16) documentation here https://virtualenv.pypa.io/en/legacy/reference.html. v20 aimes to be cli compatible with v16 baring some flags that have been deprecated for years now. https://virtualenv.pypa.io/en/legacy/reference.html#cmdoption-no-site-packages is one of this, so we've dropped it during the migration to the new major version. We're sorry if this caused issues for you, you're right we should have mentioned it somewhere in our release changelog though.

**Comment by humitos** [comment]
Hi all! We did a release of our docker image that it's used to build the documentation, to fix another issue on Friday. Since we don't pin `virtualenv` on it, it got updated and we didn't realize that there was a breaking change. I think we rolled back this update, but I will double check this tomorrow with more time (from red-discordbot builds, it seems it's already rolled back because those builds are passing)

Please, let me know if you are still experimenting this issue.

**Comment by ericholscher** [comment]
We've also released a fix for this in #6738, which was deployed yesterday. Closing this now.
